src/app/server/Mdns.cpp

```diff
@@ -148,17 +148,17 @@ bool MdnsServer::OnExpiration(uint64_t expirationMs)
 
     ChipLogDetail(Discovery, "OnExpiration - valid time out");
 
-    // reset advertising
-    CHIP_ERROR err;
-    err = chip::Mdns::ServiceAdvertiser::Instance().StopPublishDevice();
+    CHIP_ERROR err = Mdns::ServiceAdvertiser::Instance().Init(&chip::DeviceLayer::InetLayer, chip::Mdns::kMdnsPort);
```

Comment:
Could this be a good time to avoid singleton access constantly for ServiceAdvertiser, and instead inject it to MdnsServer at initialization?

Reply:
This PR is already quite a big. Can I file a followup for that?

Reply:
Yes

Reply:
https://github.com/project-chip/connectedhomeip/issues/10225